addons/binding/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/handler/PHCBridgeHandler.java

```diff
@@ -0,0 +1,693 @@
+
+            serialPort.addEventListener(this);
+            // activate the DATA_AVAILABLE notifier
+            serialPort.notifyOnDataAvailable(true);
+            serialPort.setRTS(true);
+
+            // get the output stream
+            serialOut = serialPort.getOutputStream();
+
+            sendPorBroadcast((byte) 0xFF);
+
+            byte[] b = { 0x01 };
+            for (int j = 0; j <= 0x1F; j++) {
+                serialWrite(buildMessage((byte) j, 0, b, false));
+
+            }
+            updateStatus(ThingStatus.ONLINE);
+
+            scheduler.submit(new Runnable() {
+
+                @Override
+                public void run() {
+                    processQueueLoop();
+                }
+
+            });
+
+        } catch (PortInUseException | TooManyListenersException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Could not open serial port " + serialPort + ": " + e.getMessage());
+
+        } catch (UnsupportedCommOperationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Could not configure serial port " + serialPort + ": " + e.getMessage());
+        } catch (NoSuchPortException e) {
+            StringBuilder sb = new StringBuilder();
+            @SuppressWarnings("unchecked")
+            Enumeration<CommPortIdentifier> portList = CommPortIdentifier.getPortIdentifiers();
+            while (portList.hasMoreElements()) {
+                CommPortIdentifier id = portList.nextElement();
+                if (id.getPortType() == CommPortIdentifier.PORT_SERIAL) {
+                    sb.append(id.getName() + "\n");
+                }
+            }
+            logger.error("Serial port '{}' could not be found. Available ports are:\n {}", port, sb);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Serial port '" + port + "' could not be found.");
+        }
+    }
+
+    /**
+     * Puts the available data on serial port into a buffer and transfer it to the processing method.
+     */
+    @Override
+    public void serialEvent(SerialPortEvent event) {
+        if (event.getEventType() != SerialPortEvent.DATA_AVAILABLE) {
+            return;
+        }
+
+        byte[] buffer;
+
+        try {
+            if (serialIn.available() <= 0) {
+                return;
+            }
+
+            Thread.sleep(5);
+            buffer = new byte[serialIn.available()];
+            serialIn.read(buffer);
+
+            if (messageFragment != null) {
+                processInputStream(ArrayUtils.addAll(messageFragment, buffer));
+            } else {
+                processInputStream(buffer);
+            }
+
+        } catch (IOException e) {
+            // ignore
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.